src/utils/formatDate.ts

```diff
@@ -0,0 +1,33 @@
+import { months } from './constants';
+
+const formatDate = (dateString: string): string => {
+  const isValidDate = (dateString: string): boolean => {
+    const date = new Date(dateString);
+    return !isNaN(date.getTime());
+  };
+
+  if (!isValidDate(dateString)) {
+    return '';
+  }
+
+  const date = new Date(dateString);
+  let day: number | string = date.getDate();
+  const month: string = months[date.getMonth()]!;
+  const year: number = date.getFullYear();
+  let hours: number | string = date.getHours();
+  let minutes: number | string = date.getMinutes();
+
+  minutes = minutes < 10 ? '0' + minutes : minutes;
+  day = day < 10 ? '0' + day : day;
+
+  let formattedDate = `${day} ${month} ${year}`;
+
+  if (hours !== 0 || minutes !== '00') {
```

Comment:
what if the 0:00 am ? dont you want to format then the hours to 00??

Reply:
Every time we receive a response with the time values as `00:00:00`. When we extract it from the JavaScript side, we get 0 when it was 00 for both hours and minutes. To justify why we have this condition `hours!== 0 || If the minutes value is '00'`, we append 0 after extraction. The same has been done hours after the condition. We are not displaying `0000` at the UI level, only showing other numbers except for `0000`.

Reply:
Now the minutes condition has been moved to the inside of the if block. We should not display time when both values are 0 after extraction.